ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java

```diff
@@ -266,70 +262,33 @@ private String handleCreateAsSelect(
       final String statementStr,
       final Mode mode
   ) {
-    final QuerySpecification querySpecification =
-        (QuerySpecification) statement.getQuery().getQueryBody();
-    final Query query = ksqlEngine.addInto(
-        querySpecification,
-        statement.getName().getSuffix(),
-        statement.getQuery().getLimit(),
-        statement.getProperties(),
-        statement.getPartitionByColumn(),
-        true
-    );
-    startQuery(statementStr, query, command, mode);
+    startQuery(statementStr, command, mode);
     return statement instanceof CreateTableAsSelect
         ? "Table created and running" : "Stream created and running";
   }
 
   private String handleInsertInto(
-      final InsertInto statement,
       final Command command,
       final String statementStr,
       final Mode mode) {
-    final QuerySpecification querySpecification =
-        (QuerySpecification) statement.getQuery().getQueryBody();
-    final Query query = ksqlEngine.addInto(
-        querySpecification,
-        statement.getTarget().getSuffix(),
-        statement.getQuery().getLimit(),
-        new HashMap<>(),
-        Optional.empty(),
-        false
-    );
-    startQuery(statementStr, query, command, mode);
+    startQuery(statementStr, command, mode);
     return "Insert Into query is running.";
   }
 
   private void startQuery(
       final String queryString,
-      final Query query,
       final Command command,
       final Mode mode
-  ) { 
-    if (query.getQueryBody() instanceof QuerySpecification) {
-      final QuerySpecification querySpecification = (QuerySpecification) query.getQueryBody();
-      final Relation into = querySpecification.getInto();
-      if (into instanceof Table) {
-        final Table table = (Table) into;
-        if (ksqlEngine.getMetaStore().getSource(table.getName().getSuffix()) != null
-            && querySpecification.isShouldCreateInto()) {
-          throw new KsqlException(String.format(
-              "Sink specified in INTO clause already exists: %s",
-              table.getName().getSuffix().toUpperCase()
-          ));
-        }
-      }
-    }
-
+  ) {
```

Comment:
I've added tests to check in `KsqlEngine` that this check is covered by the Engine.  No tests existed that checked this failure condition.  As far as I can tell the new engine checking covers this.

I'm assuming this checks for CSAS/CTAS statements where the stream/table already exists.